apps/wing/src/commands/pack.ts

```diff
@@ -0,0 +1,77 @@
+import * as path from "path";
+import * as fs from "fs";
+import * as cp from "child_process";
+
+export interface PackageOptions {
+  /**
+   * Directory to to save the generated package to.
+   */
+  readonly outdir: string;
+}
+
+export async function pack(options: PackageOptions): Promise<string> {
+  // check npm is installed
+  try {
+    cp.execSync("npm --version", { stdio: "ignore" });
+  } catch {
+    throw new Error(`npm is not installed. Install it from https://www.npmjs.com/get-npm`);
+  }
+
+  // check there is at least one source file
+  const sourceFiles = fs.readdirSync(process.cwd()).filter((f) => f.endsWith(".w"));
+  if (sourceFiles.length === 0) {
```

Comment:
As a library author I imagine it would be pretty common to want to have some entrypoint files (maybe for testing) alongside my other files. Types from entrypoint files would not be exported by the library.

Reply:
Sounds good